linked_list.rb

```diff
@@ -19,77 +19,232 @@ def initialize
   # method to add a new node with the specific data value in the linked list
   # insert the new node at the beginning of the linked list
   def insert(value)
-    puts "Not implemented"
+    node = Node.new(value)
+    if @head == nil
+      @head = node
+    else
+      node.next = @head
+      @head = node
+    end
+
+    # puts "Not implemented"
   end
 
   # method to find if the linked list contains a node with specified value
   # returns true if found, false otherwise
   def search(value)
-    puts "Not implemented"
+    current = @head
+    while current != nil
+      if current.data == value
+        return true
+      else
+        current = current.next
+      end
+
+    end
+    return false
+
+    # puts "Not implemented"
   end
 
   # method to return the max value in the linked list
   # returns the data value and not the node
   def find_max
-    puts "Not implemented"
+
+    current = @head
+    max = current.data

```

Comment:
Always check if the node is nil before accessing data in it. This will give a nil dereference error is @head is nil. So, check to ensure that @head is not nil before accessing .data.